doc/contribute/contributor_expectations.rst

```diff
@@ -219,11 +219,16 @@ defines the following escalation process for resolving technical disagreements.
   - The involved parties and the Assignee to be present when
     the (escalated) issue is discussed.
 
-- TSC: Assignees can escalate to the TSC voting members and get a binding
-  resolution in the TSC by adding the `tsc`_ label on the PR.
+- The assignee (maintainer) has the right to dismiss stale, unrelated or
+  irrelevant change requests by reviewers giving the reviewers a minimum of 1
+  business day to respond and revisit their initial change requests or escalate
+  to the architecture working group.
```

Comment:
It was agreed long ago and approved in the TSC, that this was the case. That is, that a maintainer in an area can dismiss reviews from collaborators or people less involved in their area. And that in that case, those who disagree have the right to escalate to the TSC if they feel the need.
The idea is that maintainers are trusted to make decisions in their area to a large degree but an escalation path exists.
The situation otherwise was one in which anybody could block progress in an area even if they were not involved in it, and when a disagreement started, there was too much arm waving and other not proper behaviors that you'd expect from politicians and not engineers, which was not good at all. We had had some pointless confrontations before this rule, which this solved.
It is very rare that reviews are dismissed due to disagreements, and even rarer that they are escalated to the TSC.

Disenpowering maintainers and forcing any disagreement to be brought to a meeting in which most have no understanding of the issue would bring us back to a bad situation, so I would very much object to that. Those not involved in a review are typically those who have no stake or knowledge, so they should not be the default decision maker.

Reply:
> I think cases that result in review dismissals should be carried on by someone unrelated to the author and reviewers.

this is exactly what the escalation process is about.

Reply:
I like the idea that maintainer are trusted and empowered to make decisions, but I think that there's few corner cases that should be considered, for instance:
- Driver changes usually falls between two (or more) maintainer areas: the subsystem and the platform, who end up assigned is pretty arbitrary, IIRC it's based on the file count but patform has priority. I think in case of disagreement the subsystem maintainer should have priority over the platform one.
- Situations where the assignee author and in the same org, the reviewer is external. It may be alright to dismiss but in principle it does not really play well with the vendor neutrality nature of the project.

Reply:
> this is exactly what the escalation process is about.

Oh you mean up to arch or tsc? Sure but then the dismissal could be up to the working group.

Reply:
> Oh you mean up to arch or tsc? Sure but then the dismissal could be up to the working group.

no, because you might dismiss some review that is blocking a PR for no good reason and the reviewer concerned might not even care or they might just accept it and move on, we do not want to make every disagreement go to a WG, the WG or TSC path only needed when disagreement continues and no way to resolve in the PR.

Reply:
> * Driver changes usually falls between two (or more) maintainer areas: the subsystem and the platform, who end up assigned is pretty arbitrary, IIRC it's based on the file count but patform has priority. I think in case of disagreement the subsystem maintainer should have priority over the platform one.

we already document that maintainer have the responsibility to re-assign. Anyone can contest wrong assignment and re-assign if it makes sense. But we also have to expect that 'maintainer' should be looking for the project and the common good by default and that maintainers will do the right thing at the end. There is no system that is perfect and we need to have some trust. 

> Situations where the assignee author and in the same org, the reviewer is external. It may be alright to dismiss but in principle it does not really play well with the vendor neutrality nature of the project.

This is the reality of the project and we have recently added the measure to at least not allow merges of PRs when submitter, reviewers are from the same org, so we have at least one pair of eyes that is not involved directly. Any other reviews from other collaborators are still and will always be welcome, but the reality here is that the vendors know their hardware best. As long as the context of a change is "local" to one vendor, wrong doing will only harm the vendor themselves. 
This become more critical for "common" areas, and we already request that more eyes from different organisation are part of the review.

Reply:
> [...] but the reality here is that the vendors know their hardware best. As long as the context of a change is "local" to one vendor, wrong doing will only harm the vendor themselves.

That sounds like a situation where having external contributions would improve the overall result. A feedback from someone unfamiliar with the specific hardware may highlight documentation gaps that would otherwise be missed. Consciously accepting harmful changes just because it's withing a vendor set of drivers also does not strike me as ideal.

Anyway, I think I may be misreading this, is the intended sequence like:
- assignee dismisses the review
- the reviewer has 1 day to follow up

or
- the assignee asks the reviewer to clarify or escalate to arch
- the review gets dismissed if no response after 1 day

?

Reply:
> That sounds like a situation where having external contributions would improve the overall result.

Yes of course and sometimes it happens which is unbelievable but this is realistically the situation where there is the least money and least interest to perform "outsider" reviews.

> Consciously accepting harmful changes...

Strike "consciously".

Reply:
> Anyway, I think I may be misreading this, is the intended sequence like:
> 
> * assignee dismisses the review
> * the reviewer has 1 day to follow up

this.

There are many other steps and things that can be done before a review is dismissed, i.e. discussion in the PR itself, discord, the dev-review meeting etc or even the arch. WG if the issue at hand is at the level we want to have in this meeting, but ultimately and if all options have been considered and tried, the assignee can dismiss with a justification and eventually go through the escalation process if requested.

Reply:
> There are many other steps and things that can be done before a review is dismissed, i.e. discussion in the PR itself, discord, the dev-review meeting etc or even the arch. WG if the issue at hand is at the level we want to have in this meeting, but ultimately and if all options have been considered and tried, the assignee can dismiss with a justification and eventually go through the escalation process if requested.

Ok, I missed out that this was the last item in the escalation list and already after the "talk to each other" and "talk to each other in a working group" steps. Yeah I think this is fine then, though it's odd that the working group item is "optionally resolve...". I think a reviewer should  be given the chance to ask to bring up a PR in a working group before their review is dismissed. Maybe drop "optionally" from the previous step?

Reply:
Everything LGTM, but to @fabiobaltieri concerns, this section actually starts with resolving issues in the PR, then it says "optionally" take the conversation to the dev or arch meetings. Might I suggest that we remove the word "optionally" and just make the steps:

1. Work within the PR and community infrastructure (discord) to resolve the PR
2. Bring to the dev/arch meeting to discuss

Let the discussion results from the dev/arch be the trigger to decide if the review should be dismissed or not with the assignee being able to do the dismissal?

Reply:
dev-review meeting is not part of the escalation, it is just a discussion forum and artch WG meeting should not be used to discuss every single disagreement in PRs, the backlog of technical discussion of new subsystems APIs etc is already long enough, this just does not scale, also, not everyone can attend those meetings. So escalation path should be short we defined and should not involve 3 meetings (dev-review, arch wg, TSC...).

Anyways, I added more updates, please take another look.

Reply:
Thanks I think it's a bit more clear now.